src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -869,5 +897,51 @@ public override int GetHashCode()
         {
             return !(left == right);
         }
+
+#if DEBUG
+        internal string Dump()
+        {
+            return TreeDumper.DumpCompact(Dump(this));
+
+            TreeDumperNode Dump(Conversion self)
+            {
+                var sub = new System.Collections.Generic.List<TreeDumperNode>();
+
+                if ((object)self.Method != null)
+                {
+                    sub.Add(new TreeDumperNode("method", self.Method.ToDisplayString(), null));
+                }
+
+                if ((object)self.DeconstructionInfo != null)
+                {
+                    sub.Add(new TreeDumperNode("deconstructionInfo", null,
+                        new[] { BoundTreeDumperNodeProducer.MakeTree(self.DeconstructionInfo._invocation)}));
+                }
+
+                var underlyingConversions = self.UnderlyingConversions;
+                if (!underlyingConversions.IsDefaultOrEmpty)
+                {
+                    sub.Add(new TreeDumperNode($"underlyingConversions[{underlyingConversions.Length}]", null,
+                        underlyingConversions.SelectAsArray(c => Dump(c))));
+                }
+
+                return new TreeDumperNode("conversion", self.Kind, sub);
+            }
+        }
+#endif
+    }
+
+    /// <summary>Stores all the information from binding for calling a Deconstruct method.</summary>
+    internal class DeconstructionInfo
+    {
+        internal DeconstructionInfo(BoundExpression invocation, BoundDeconstructValuePlaceholder inputPlaceholder,
+            ImmutableArray<BoundDeconstructValuePlaceholder> outputPlaceholders)
+        {
+             (_invocation, _inputPlaceholder, _outputPlaceholders) = (invocation, inputPlaceholder, outputPlaceholders);
+        }
+
+        internal BoundExpression _invocation;
+        internal BoundDeconstructValuePlaceholder _inputPlaceholder;
```

Comment:
Since we have performed the binding for invocation during initial binding anyways, it seems wasteful and also more work (to channel the information as a MethodSymbol and then reconstruct an invocation during lowering). We can discuss as a further refinement.